specutils/fitting/fitmodels.py

```diff
@@ -212,6 +212,44 @@ def _fit_lines(spectrum, model, fitter=fitting.LevMarLSQFitter(),
     return fit_model
 
 
+def _convert_and_dequantify(poss_quantity, dispersion_unit, dispersion, flux_unit):
+    """
+    This method will convert the ``poss_quantity`` value to the proper
+    dispersion or flux units and then strip the units.
+
+    If the ``poss_quantity`` is None, or a number, we just return that...
+
+    Note: This method can be removed along with most of the others here
+          when astropy.fitting will fit models that contain units.
+
+    """
+
+    if poss_quantity is None or isinstance(poss_quantity, (float, int)):
+        return poss_quantity
+
+    if hasattr(poss_quantity, 'quantity') and poss_quantity.quantity is not None:
```

Comment:
I think my comment above might have better been here.  I think I meant a leading 
```
if isinstance(poss_quantity, Quantity):
    q = poss_quantity
elif ...
```

Reply:
I made a small change to include that check.